posts/vue-svg-symbol-sprites.md

```diff
@@ -0,0 +1,177 @@
+But there is more convenient way by using a Webpack loader which allows you to use your original `.svg` files just by importing them without any modifications or copy-pasting parts of them around. It also provides a simple way to fetch the sprite image and cache it in localStorage, which makes sense for sprite images of a certain size.
+
+I use Vue.js for both work and my side-project [tidyMind][tidymind-intro] and implemented the following solution into both of them. It should be easily adjustable for other frameworks / libraries like React or Angular as well.
+
+The plugin and loader combination I use is [svg-symbol-sprite-loader][svg-symbol-sprite-loader].
+
+This is how to set it up with Vue:
+
+## Install dependency
+
+```
+npm i -D svg-symbol-sprite-loader
+```
+
+## Webpack configuration
+
+You will need a Webpack configuration which uses `html-webpack-loader`. This is necessary because the plugin uses a hook of html-webpack-loader to insert the filename of the generated sprite image into the HTML, to be picked up by the script which loads and caches it later.
+
+```
+const SVGSymbolSprite = require('svg-symbol-sprite-loader')
+
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.svg$/,
+        use: [
+          {
+            loader: 'svg-symbol-sprite-loader',
+            options: {
+              symbolId: filePath => `icon-${path.basename(filePath, '.svg')}`,
+            },
+          },
+        ],
+      },
+    ],
+  },
+  plugins: [
+    // ⚠️ must be included _after_ the HtmlWebpackPlugin
+    new SVGSymbolSprite.Plugin({
+      filename: `icon-sprite${process.env.NODE_ENV === 'production' ? '.[chunkhash]' : ''}.svg`
+    }),
+  ],
+}
+```
+
+The `symbolId` option is optional, but when you omit it the `id` attributes of the `symbols` will just be the filenames of your SVGs (`unicorn.svg` becomes `id="unicorn"`). This can cause duplicate IDs in the document and [lead to bugs][collisions].
+
+You can also consult the [official documentation][webpack-config] for this part.
+
+## Load and cache the sprite file
+
+[svg-symbol-sprite-loader][svg-symbol-sprite-loader] conveniently provides a function which you just need to call to fetch and cache the sprite image:
+
+```
+import svgSymbolSpriteLoader from 'svg-symbol-sprite-loader'
+
+svgSymbolSpriteLoader({ useCache: process.env.NODE_ENV === 'production' })
+```
+
+Just be aware that this uses the [Fetch API][fetch-api] which is not supported by Internet Explorer.
+
+If you need to support IE, you can either import a [fetch polyfill][fetch-polyfill] or copy and adapt [the function][icon-sprite-loader] to the HTTP client you use. The latter is what I did for my work projects where we support IE, already have bundled [Superagent][superagent] as a HTTP client and did not want to increase the bundle size by including a polyfill which is otherwise not needed.
```

Comment:
... to the HTTP client you use. **I have done the latter for my work projects. Since IE is supported there and Superagent is already bundled as HTTP client, I didn't want to increase the bundle size by an otherwise not needed polyfill.**